README.md

```diff
@@ -1,21 +1,148 @@
+- [ℹ️ About](#%E2%84%B9%EF%B8%8F-about)
+- [🤝 Contributing](#%F0%9F%A4%9D-contributing)
+- [⚖️ License](#%E2%9A%96%EF%B8%8F-license)
+
+## 💡 What does P2P Editor do?
+
+P2P editor is a code editor that works in the browser which lets you share live coding sessions.
+
+### How it works?
+
+You star a new session with an empty source code file and let other users join the session. All connected users will see all your code changes reflected in their read only editor.
+
+The functionalities are limited for now, but we have a roadmap to make P2P editor awesome. If you want to get involved take a look of [Contributing](#-contributing) section. 
+
+## 🤔 Why P2P?
+
+We have remove centralized servers, so by doing this, we have the good and the bad of that:
+
+### The good part
+- Direct communication: reduce latency and bandwidth.
+- No servers: no cloud, no scaling, no deploy, no devops...
+- No SasS:
+  - Availability.
+  - No usage restriction.
+  - No unknown (malicious or not) usage of your data.
+
+### The bad part
+- Data consistency.
+- Network partition / topology.
+- Peer discovery and connection establishment.
+- Authentication / Authorization.
+- Security and privacy.
+
+We have a particular interest in P2P application development. You can read more in [About](#-about) section.
+
+## 🏗️ Architecture
+
+![Arquitecture Diagram](assets/architecture_diagram.jpg)
+
+Parts of P2P editor:
+- Code editor: [Ace](https://github.com/ajaxorg/ace)
+- Database: append only log ([hypercore](https://github.com/mafintosh/hypercore))
+- Communication: WebRTC RTCDataChannel ([webrtc-swarm](https://github.com/mafintosh/webrtc-swarm))
+- Peer discovery: WebRTC signaling server ([signalhub](https://github.com/mafintosh/signalhub))
+- Storage: RAM
+
+Workflow:
+1. Peers are discovered via signaling server.
+2. For each change made in the editor, a delta representing that change is appended to hypercore database.
+3. Hypercore database is constantly replicating between all peers via WebRTC RTCDataChannel.
+4. When a peer receives a delta it is applied to its editor instance.
+
+Data consistency: [Eventual consistency](https://en.wikipedia.org/wiki/Eventual_consistency)
+## 🚀 Environment setup
+
+### 🐳 Docker environment
+
+* Clone this repository: `git clone https://github.com/CodelyTV/p2p-editor`
+* Move to your project folder: `cd p2p-editor`
+* Copy the default environment variables: `cp .env.dist .env`
+* Start the services: `docker-compose up -d`
+* Go to [http://localhost:8000](http://localhost:8000)
+
+### 🎰 Local environment
+
+
+* Clone this repository: `git clone https://github.com/CodelyTV/p2p-editor`
+* Move to your project folder: `cd p2p-editor`
+* Copy the default environment variables: `cp .env.dist .env`
+* Install dependencies: `yarn install` or `npm install`
+* Start dev server: `yarn start` or `npm start`
+* Go to [http://localhost:8000](http://localhost:8000)
+
+
+### ✅ Run the tests
+
+`yarn test` or `npm run test `
```

Comment:
We should add how to run the test from inside the docker. [Example](https://github.com/CodelyTV/cqrs-ddd-php-example/commit/61d2ac2687aedafb69079a131729c73cf7684821#diff-04c6e90faac2675aa89e2176d2eec7d8R78)

Reply:
This is the first time I see this pattern of using Make in conjunction with docker. Cool!

For now i'll replace yarn/npm commands with the corresponding docker-compose command `docker-compose run p2p-editor test`.

Reply:
I've created an issue for this https://github.com/CodelyTV/p2p-editor/issues/25